### DISCUSSION TITLE
Is there a way to use service mesh micro with a durable consumer?

### DISCUSSION THREAD

**Discussion by cr1cr1** [nats-io/nats.go#1519] [Q&A] [answered]
Please bear with me as I am a complete beginner to nats. Learned about this awesome project some years ago, but just now starting to use it.

In the meantime, there are new APIs, 2nd major version, great videos about how it works and how to use it. Great times!

I am trying to use the service mesh library and it works just fine as per the examples.

I even managed to have the messages persisted with JetStream using the new APIs, they reach the service handler but remain unprocessed in a consumer.

Just using req/response is not enough for some use cases, e.g. a simple CRUD interface for objects like a User and would be ideal that a service would pick up async messages, in order - instead of replying immediately.

Looking in the source code does not seem to be possible to use a durable consumer with the micro approach, does it?

Thank you!

**Comment by Jarema** [comment, marked as answer]
Hey!

Yes, you are right.
Service API is based on Core NATS only.

Keep in mind that Service API is still Core NATS, so you should be able to mix and match those two, including JetSteram in your code and APIs.
